### ISSUE TITLE
Setting global.zeebe causes startup to fail

### ISSUE THREAD

**Issue by bastianeicher** [zeebe-io/zeebe-cluster-helm#63]
I wish to use the value `global.zeebe` to name the Kubernetes resources `zeebe` rather than `zeebe-zeebe`.

## Works in old version

In version 0.0.89 of the chart this works fine:

```
> helm install zeebe zeebe/zeebe-cluster --version 0.0.89 --set global.zeebe=zeebe

> kubectl get pods
NAME                             READY   STATUS
elasticsearch-master-0           1/1     Running
elasticsearch-master-1           1/1     Running
elasticsearch-master-2           1/1     Running
zeebe-0                          1/1     Running
zeebe-1                          1/1     Running
zeebe-2                          1/1     Running
zeebe-gateway-5f89d47657-pl8jw   1/1     Running
```

## Breaks in current version

In the latest versions of the chart this causes the startup to fail:

```
> helm install zeebe zeebe/zeebe-cluster --version 0.0.99 --set global.zeebe=zeebe

> kubectl get pods
NAME                             READY   STATUS
elasticsearch-master-0           1/1     Running
elasticsearch-master-1           1/1     Running
elasticsearch-master-2           1/1     Running
zeebe-0                          0/1     CrashLoopBackOff
zeebe-1                          0/1     CrashLoopBackOff
zeebe-2                          0/1     CrashLoopBackOff
zeebe-gateway-5657db69c4-bzfvw   1/1     Running

> kubectl logs zeebe-0
[...]
***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to bind properties under 'zeebe.broker.gateway.network.port' to int:

    Property: zeebe.broker.gateway.network.port
    Value: tcp://10.0.96.44:9600
    Origin: "zeebe.broker.gateway.network.port" from property source "systemProperties"
    Reason: failed to convert java.lang.String to int
```

I am using Helm 3.

**Comment by salaboy** [comment]
@bastianeicher thanks for reporting.. can you try version 100?

**Comment by salaboy** [comment]
@bastianeicher thanks for reporting.. i can confirm that this also fails in verison 100.. 
It seems that something is failing while replacing the global.. In general we should use the release name to name our cluster.. that is why the global default is set to` {Release.Name}-zeebe`
Is there any reason why you want to override that? I am not entirely sure why you want to do that.

**Comment by salaboy** [comment]
In your case.. the cluster is called `zeebe-zeebe` but the first `zeebe` is the release name

**Comment by bastianeicher** [comment]
@salaboy: I am deploying Zeebe in its own Kubernetes namespace and only have one instance of Zeebe in total. Therefore the only logical Helm release name for me is `zeebe`. The resulting cluster name `zeebe-zeebe` seems a little redundant to me.

More specifically, when I specify the Zeebe hostname for other services in the cluster to connect to I would like it to be `zeebe-gateway.zeebe.svc.cluster.local` rather than `zeebe-zeebe-gateway.zeebe.svc.cluster.local`.

So using the Helm release name as the Zeebe cluster name is fine for me, but I would prefer to have it without an additional `-zeebe` appended at the end.

**Comment by salaboy** [comment]
@bastianeicher I do understand but that is not a typical use of HELM, you can use the namespace name for your release, so it is clear to which zeebe cluster you are connecting to.
If we remove the `-zeebe` and your release is called `whatever` that is what the name of the zeebe cluster service will be, which is extremely confusing in my opinion. 

I bet that this issue can be fixed, but it will take some time to test, and it will be confusing for users that want to use Helm with release names. If you feel like sending a PR with a solution that works for all the mentioned use cases, please do. I will happily review it. but for now, this is a no-fix

**Comment by bastianeicher** [comment]
I agree that not having `zeebe` in the pod names at all would be confusing. However, there is a common pattern used by many charts to address this:

- combine the Release name and Chart name,
- unless the Release name already contains the Chart name,
- unless an explicit override is specified.

For example using the `mariadb` Chart:

**Combine the Release name and Chart name**

```
> helm install whatever stable/mariadb
> kubectl get pods
NAME                        READY   STATUS
whatever-mariadb-master-0   1/1     Running
[...]
```

**Release name already contains the Chart name**

```
> helm install mariadb stable/mariadb
> kubectl get pods
NAME                        READY   STATUS
mariadb-master-0            1/1     Running
[...]
```

**Explicit override is specified**

```
> helm install whatever stable/mariadb --set fullnameOverride=whatever
> kubectl get pods
NAME                        READY   STATUS
whatever-master-0           1/1     Running
[...]
```

The Zeebe chart already contains the logic for this:
https://github.com/zeebe-io/zeebe-cluster-helm/blob/968b147c82541f992f7da6e969ab05ad183b8825/zeebe-cluster/templates/_helpers.tpl#L14-L25

**Comment by salaboy** [comment]
@bastianeicher you are right, and I like that approach.. the more aligned than we are with other helm charts the better, would you send a PR with the change?

**Comment by kjozsa** [comment]
I confirm this issue happening with 0.23.1, I have just ran into the same problem.

**Comment by npepinpe** [comment]
With https://github.com/zeebe-io/zeebe-cluster-helm/pull/79, this may have been solved. I'll confess I wasn't aware of this issue, but the names should be standardized around the existing templates. That said, I still reused `global.zeebe` because I didn't want to break and remove it, though I think using the more standard `Release.Name` or `Values.fullnameOverride` would be better in my opinion.

Happy to modify it so we use this pattern if others agree

**Comment by salaboy** [comment]
@npepinpe the use of the global is required to make it work with the operate chart if they are used without having a parent chart such as `zeebe-full` as soon as these convinations work.. I am fine with refactoring the names. I do believe that this is an improvement for usability, but we need to make sure that we don't break other charts.
